src/DetailsView/components/details-view-content.tsx

```diff
@@ -145,16 +147,19 @@ export const DetailsViewContent = NamedFC<DetailsViewContentProps>('DetailsViewC
                 props.storeState,
             );
 
+        const selectedTestKey = props.deps.getProvider().forType(selectedTest)?.key;
```

Comment:
I think we should use the configuration factory instead here: the details-view-content is a general (i.e. not only for assessment) space. This should also remove the need for the `?`, I think.